Switching to Arch Linux

Mange opened the pull request:
I'm trying to switch to Arch Linux. This PR will document my journey until it's usable enough to do a full switch over.

**Parts:**

- [x] Add setup script.
- [x] Automatically install AUR tools for community-built software.
- [x] Install basics of dev tools (git, neovim, etc.)
- [x] Install basics of i3, etc.
- [x] Install all software from the Ubuntu lists.
- [x] Get bluetooth working well.
- [x] Get SSH agent, GPG agent working.
- [x] Style lightdm.
- [x] Verify audio solution (~vanilla Alsa or~ PA ~?~).
- [x] Add bootstrap script (to run as root after installing the dist).
  * Create my own user, etc.
  * Install git and clone the dotfiles, then run the platform script.
- [x] Dotfiles for termite.
- [x] ~Build a system to install my own projects using AUR rather than the current implementation.~ Out of scope. Not needed currently!
- [x] Power/energy settings.
- [x] bilquis software (my laptop)
  - [x] System76 driver, trackpad, etc.
- Extra software
  - [x] Termite.
  - [x] Docker.
  - [x] RVM via package manager.
  - [x] Automatically install and switch to latest Ruby on bootstrapping.
- Cleanup
  - [x] Remove vendor code for i3-gaps, fonts, etc. AUR has it now.
  - [x] Remove ubuntu scripts.
  - [x] Remove barcommands.

## Things to do later

- Fix icons in polybar (Nerdfonts should be used now)

Mange committed:
Start working on Arch Linux platform script

Mange committed:
Add more to the Arch setup script

Mange committed:
Arch: termite replaces gnome-terminal

Mange committed:
Arch: Add settings to skip certain sections

Mange committed:
Arch: Add neovim section

Mange committed:
Arch: Install vlc and playerctl

Mange committed:
Arch: Install AUR software automatically

Mange committed:
Arch: Install pavucontrol

Mange committed:
Disable spellcheck in package lists

Mange committed:
Make handle-failure render errors better

Mange committed:
Arch: Enable bluetooth support

Mange committed:
Arch: Install some missing bar applets

Mange committed:
Improve install flow for AUR

Now each package is installed one at a time, so if many new packages are
pending and one of them fails to compile, then the rest of the packages
are still compiled and installed.

Next time you re-run the command only the failing packages from last
time will be compiled and installed.

Mange committed:
Arch: Install remaining software from Ubuntu apts lists

Mange committed:
Arch: Add a lot of fortune packages

Mange committed:
Arch: Configure LightDM

Mange committed:
Add dotfiles for termite terminal emulator

Mange committed:
Arch: Automatically install docker and docker-compose

Mange committed:
Arch: Install RVM via AUR instead of manually

Mange committed:
Arch: Install YakYak

Mange committed:
Arch: Configure GPG to automatically fetch keys

Mange committed:
Arch: Install Slack and Discord

Mange committed:
Arch: Show power menu on DM screen

Mange committed:
Arch: Install dropbox

Mange committed:
Arch: Add bootstrapping script

Mange committed:
Rewrite README

Mange committed:
Arch: Install icon fonts via Nerd Fonts (AUR)

This removes a whole lot of custom stuff I have to maintain myself in
this repo. :-)

Mange committed:
Arch: Add updates section; install AUR updates too

Mange committed:
Remove vendor recipies; AUR has everything now

rofi-lpass was actually my own piece of software, but I'm no longer
using LastPass for my password manager (I'm using Bitwarden now). So it
was unused anyway...

Mange committed:
Remove Ubuntu platform scripts

Mange committed:
Stop auto-installing barcommands

I'm not using these anymore since I switched to polybar. It's just
adding extra complexity to the setup phase right now[1].

[1]: For example, they are cloned using SSH to Github, so they will
always fail on the first few setups.

Mange committed:
Arch: Support apps requesting root in xorg

Mange committed:
Arch: More FS support

Mange committed:
Arch: Don't sort AUR packages

Order in aur.txt should be significant so higher-priority software is
installed first.

Mange committed:
Fix some shellcheck warnings

Mange committed:
Arch: Start network and lightdm on boot

(Just in case NetworkManager is not set to start on boot already. It
really should be at this point since the script got this far. But who
knows, right? Doesn't hurt to include it too.)

Mange committed:
Arch: Install htop

Mange committed:
Remove di-download script

I usually listen to Spotify over Digitally Imported nowadays anyway...

Mange committed:
Arch: Install Deluge on morbidus

skanev commented:
Oh noes!

Why though? I did dual Mac-Arch for a while and was immensely happy when I killed Arch and went full-time Mac. I know you've been attached to your Linux for quite a while now, but what made you switch away from Mac (apart from the new shitty laptops :))

Mange commented:
I switched away from macOS around a year ago. I had not been impressed for years and every new release was a let-down of mediocrity. I read that the macOS team became a subdivision of iOS and I kept longing for software that wasn't available on Mac, like good window managers and such.

Then I saw the touch bar and the butterfly keyboards. I vowed to never use MacBooks again.

Then I saw people around me use the new MacBooks and never liking it. I was on a two year old release of macOS at that point (no need release warranted "upgrading" the OS. "A new font on menus? Whopti-fucking-do!")

So I just switched at work then too. Tried dual booting first but the MacBook hardware is too non-standard (they gut the ACPI signals for power management and don't use standard interfaces to GPUs, for example) so that sucked.

Switched to a new laptop and have never been happier. :)

This PR is about switching from Ubuntu to Arch, which so far has been smooth. Having setup scripts and dotfiles helps to smooth over the "lol tinkering" part of it. I had another reset on my desktop while setting this up and everything installed and configured itself without issue. Will switch to it on all machines and merge this PR in the coming weeks.

(You gotta have to try i3. You might go back to Linux again! I use home row for almost everything now.)

skanev commented:
> (You gotta have to try i3. You might go back to Linux again! I use home row for almost everything now.)

I doubt it – I'm doing iOS these days, so my investment in Apple has gone higher. :)

Otherwise, thanks for the update, moving from Ubuntu to Arch is definitely a good idea in my book :)

Mange commented:
Ha ha, I don't envy you 😆

Mange committed:
Arch: Install xsel

Mange committed:
Arch: Install firefox-developer-edition

Mange committed:
Fix rust bootstrapping code

"rustup" is not on ENV until dotfiles are already installed, so manually
look for Cargo/Rustup PATH when setting it up.

Mange committed:
Arch: Add bilquis host-specific software

Mange committed:
Arch: Load RVM from Pacman-based install directory

Mange committed:
Arch: Install deja-dup backup utility

Mange committed:
Arch: Automatically install Ruby

Mange committed:
Swap synaptics with libinput

Mange committed:
Arch: Lock screen on suspend, etc.

Using light-locker was much simpler than to manually creating Systemd
service files, registering them and so on. Sure, it's not that hard,
but installing them means a lot more custom code here than just
installing an already-built package that does pretty much exactly what I
want already. :o)
